pygeos/transform.py

```diff
@@ -0,0 +1,74 @@
+from . import ufuncs, Geometry
+import numpy as np
+
+__all__ = ["apply", "transform"]
+
+
+def apply(geometry, transformation):
+    """Apply a function to the coordinates of a geometry.
+
+    Parameters
+    ----------
+    geometry : Geometry or array_like
+    transformation : function
+        A function that transforms a (N, 2) ndarray of float64 to another
+        (N, 2) ndarray of float64.
+
+    Examples
+    --------
+    >>> apply(Geometry("POINT (0 0)"), lambda x: x + 1)
+    <pygeos.Geometry POINT (1 1)>
+    >>> apply(Geometry("LINESTRING (2 2, 4 4)"), lambda x: x * [2, 3])
+    <pygeos.Geometry LINESTRING (4 6, 8 12)>
+    >>> apply(None, lambda x: x) is None
+    True
+    >>> apply([Geometry("POINT (0 0)"), None], lambda x: x).tolist()
+    [<pygeos.Geometry POINT (0 0)>, None]
+    """
+    geometries = np.asarray(geometry, dtype=np.object)
+    coordinates = ufuncs.get_coordinates(geometries)
+    coordinates = transformation(coordinates)
+    geometries = ufuncs.set_coordinates(geometries, coordinates)
+    if geometries.ndim == 0:
+        return geometries.item()
```

Comment:
Only do this if the input was not a 0dim array?

Reply:
Yes, that should still be fixed